src/lib/components/NewLayeredMap/layers/L.imageWebGLOverlay.js

```diff
@@ -139,6 +122,9 @@ L.ImageWebGLOverlay = L.Layer.extend({
         if(!options.colorScale) {
             this._colormapUrl = null;
         } else {
+            if(options.colorScale instanceof Object && this.options.colorScale instanceof Object) {
```

Comment:
I probably was not clear enough (😅), but I think I mentioned that you have to move this if-statement, as options.colorScale and this.options.colorScale has already been merged here, and is the same. 

Move the if-statement above the `Util.setOptions(...)` and add the following:
```
if(options.colorScale && options.colorScale instanceof Object && this.options.colorScale instanceof Object) {
 options.colorScale = Object.assign(this.options.colorScale, options.colorScale)
}
```

Reply:
Can you also add this logic to the shader as well? 😮